openmicroscopy/openmicroscopy
# OMERO.web deploy with WSGI

Django’s primary deployment platform is WSGI, the Python standard for web servers and applications.

To test:
- **nginx + gunicorn**:
  -  pip install gunicorn
  - bin/omero config set omero.web.debug False 
  - bin/omero config set omero.web.application_server "wsgi-tcp"
  - optional:
    - bin/omero config set omero.web.static_url '/gunicorn/static/'
    - bin/omero config set omero.web.prefix '/gunicorn'
    - bin/omero config set omero.web.application_server.host "0.0.0.0"
  - development (there will be more features added):
    - copy site configuration from bin/omero web config gunicorn-development
  - production:
    - copy site configuration from bin/omero web config gunicorn
  - bin/omero web start
  - restart/reload nginx

- **apache + mod_wsgi**:
  - install mod_wsgi https://code.google.com/p/modwsgi/wiki/InstallationInstructions (OSX ```brew install mod_wsgi```)
  - bin/omero config set omero.web.debug False 
  - bin/omero config set omero.web.application_server "wsgi"
  - optional:
    - bin/omero config set omero.web.static_url '/apache/static/'
    - bin/omero config set omero.web.prefix '/apache'
  - copy site configuration from bin/omero web config apache-wsgi
  - depends on your OS: WSGISocketPrefix run/wsgi or WSGISocketPrefix /var/run/wsgi, see https://code.google.com/p/modwsgi/wiki/ConfigurationIssues
  - bin/omero web syncmedia
  - restart/reload apache

cc: @manics
